embassy/src/time/duration.rs

```diff
@@ -38,6 +38,15 @@ impl Duration {
         }
     }
 
+    /*
+        NOTE: us delays may not be as accurate
+    */
+    pub const fn from_us(millis: u64) -> Duration {
```

Comment:
- rename to `from_micros` for consistency with rust's std
- Arg name should be `micros`, not `millis`

Reply:
Can you please add as_micros for completeness too?